src/flask/cli.py

```diff
@@ -863,37 +863,66 @@ def run_command(
     )
 
 
+shell_choices = ("python", "ipython")
+
+
 @click.command("shell", short_help="Run a shell in the app context.")
+@click.option(
+    "-i",
+    "--interface",
+    type=click.Choice(shell_choices),
+    default="python",
+    help="Which interactive shell to use.",
+)
 @with_appcontext
-def shell_command():
+def shell_command(interface):
     """Run an interactive Python shell in the context of a given
     Flask application.  The application will populate the default
     namespace of this shell according to it's configuration.
 
     This is useful for executing small snippets of management code
     without having to manually configure the application.
     """
-    import code
     from .globals import _app_ctx_stack
 
     app = _app_ctx_stack.top.app
-    banner = (
-        f"Python {sys.version} on {sys.platform}\n"
-        f"App: {app.import_name} [{app.env}]\n"
-        f"Instance: {app.instance_path}"
-    )
     ctx = {}
+    ctx.update(app.make_shell_context())
 
-    # Support the regular Python interpreter startup script if someone
-    # is using it.
-    startup = os.environ.get("PYTHONSTARTUP")
-    if startup and os.path.isfile(startup):
-        with open(startup) as f:
-            eval(compile(f.read(), startup, "exec"), ctx)
+    if interface == shell_choices[0]:
+        banner = (
```

Comment:
I decided to split up the banner variable into 2 different strings, to be able to also display the IPython version easily. If you think that is not necessary this could be simplified.